### PR TITLE
[IMPROVEMENT] Activity indicator on web view OAuth

### LINKED ISSUE #904: [IMPROVEMENT] Add loading indicator on OAuth
If we tap on Facebook login, for example, the webview takes a few seconds to load FB login, but shows a white screen instead of a loading indicator.

We need to add something like this to track loading state:
https://github.com/react-native-community/react-native-webview/blob/master/docs/Reference.md#onloadprogress

### PR DESCRIPTION
**Pull request by weijia-yu**
@RocketChat/ReactNative

Closes #904

Currently users can log in using facebook, github and gitlab, when the webview starts to load, it will show the activity indicator, if it succeeds or fails to load, it will dismiss the activity indicator

ezgif-4-7bf5244422af

### CODE REVIEW

**Comment by diegolmello** [comment]
@weijia-yu Nice! Can you add a gif?

**Comment by weijia-yu** [comment]
> @weijia-yu Nice! Can you add a gif?

You mean show a gif instead of a loading bar? What gif file do you have in mind?

**Comment by diegolmello** [comment]
> > @weijia-yu Nice! Can you add a gif?
> 
> You mean show a gif instead of a loading bar? What gif file do you have in mind?

Nope. A gif of your PR in the description.

**Comment by weijia-yu** [comment]
@diegolmello here is the gif
ezgif-4-7bf5244422af

**Comment by diegolmello** [comment]
@weijia-yu Very cool! Thanks :)

**Comment by weijia-yu** [comment]
@djorkaeffalexandre @diegolmello can someone review and approve it?

File: app/views/OAuthView.js [hunk 1]
```diff
@@ -29,8 +41,10 @@ export default class OAuthView extends React.PureComponent {
 	constructor(props) {
 		super(props);
 		this.state = {
-			logging: false
+			logging: false,
+			loading: false
 		};
+
```
Comment (djorkaeffalexandre) [inline app/views/OAuthView.js]: remove this empty line.

Comment (weijia-yu) [reply app/views/OAuthView.js]: fixed

**Comment by weijia-yu** [comment]
@djorkaeffalexandre any clue why the tests take so long?

**Comment by weijia-yu** [comment]
@diegolmello do you know why the testing is still in process?

**Comment by diegolmello** [comment]
> @diegolmello do you know why the testing is still in process?

It's not. It's on hold. I'm not going to start it, because there's no need to trigger all e2e for this change.
I'll review and merge your PR in a few days.

**Pull request merged**